### PR TITLE
feat: add `coder_git_auth` data source

### PR DESCRIPTION
**Pull request by kylecarbs**
This data source enables template authors to require git authentication for specific providers on workspace build.

In Coder, we'll parse these when creating a template version and validate Git authentication before initiating a workspace build.

### CODE REVIEW

File: examples/data-sources/coder_git_auth/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+provider "coder" {
+
```
Comment (mtojek) [inline examples/data-sources/coder_git_auth/data-source.tf:2]: nit: empty line

File: provider/gitauth_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package provider_test
+
+import (
+	"testing"
+
+	"github.com/coder/terraform-provider-coder/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/stretchr/testify/require"
```
Comment (mtojek) [inline provider/gitauth_test.go:10]: nit: mixed imports

**Review by mtojek — APPROVED** [review]
I pointed 2 nit-picks, but it LGTM.

File: provider/gitauth_test.go (hunk 2 above)
Comment (kylecarbs) [reply provider/gitauth_test.go:10]: Ahh, ty

**Pull request merged**